code/modules/cm_marines/orbital_cannon.dm

```diff
@@ -398,10 +401,40 @@ var/list/ob_type_fuel_requirements
 		return TRUE
 	return FALSE
 
+// proc designed for handling ob camera shakes, takes the target location as input and calculates camera shake based off user location.
+/obj/structure/ob_ammo/warhead/proc/handle_ob_shake(turf/epicenter)
+
+	var/radius_size = 30
+
+	for(var/mob/living/user in urange(radius_size, epicenter))
+
+		var/distance = get_accurate_dist(get_turf(user), epicenter)
+
+		var/distance_percent = ((radius_size - distance) / radius_size)
+
+		var/total_shake_factor = abs(max_shake_factor * distance_percent)
+
+		// it's of type cluster.
+		if(!max_knockdown_time)
+			shake_camera(user, 0.5, total_shake_factor, frequency)
+			continue
+
+		shake_camera(user, total_shake_factor, frequency)
```

Comment:
Are your parameters correct here? total_shake_factor is currently positionally set to be steps, and frequency is positionally set to be strength. Use named arguments e.g. `strength = total_shake_factor`.